### PR TITLE
Precompute and pass in pdists into kmeans init

### LINKED ISSUE #47: Kmeans++ dists are being recomputed for each random seed, when it can be cached
https://github.com/YosefLab/scib-metrics/blob/5f78a716e7c8036e785fa5ae9c024894153cf316/src/scib_metrics/utils/_kmeans.py#L49

### PR DESCRIPTION
**Pull request by martinkim0**
Closes #47

### CODE REVIEW

File: src/scib_metrics/utils/_kmeans.py [hunk 1]
```diff
@@ -11,7 +11,7 @@
 from ._utils import get_ndarray, validate_seed
 
 
-def _initialize_random(X: jnp.ndarray, n_clusters: int, key: jax.random.KeyArray) -> jnp.ndarray:
+def _initialize_random(X: jnp.ndarray, n_clusters: int, p_dists: jnp.ndarray, key: jax.random.KeyArray) -> jnp.ndarray:
```
Comment (martinkim0) [inline src/scib_metrics/utils/_kmeans.py:14]:
```suggestion
def _initialize_random(X: jnp.ndarray, n_clusters: int, pdists: jnp.ndarray, key: jax.random.KeyArray) -> jnp.ndarray:
```

File: src/scib_metrics/utils/_kmeans.py [hunk 2]
```diff
@@ -120,6 +122,7 @@ def fit(self, X: np.ndarray):
         return self
 
     def _fit(self, X: np.ndarray):
+        self._pdists = pdist_squareform(X)
```
Comment (adamgayoso) [inline src/scib_metrics/utils/_kmeans.py:125]: this is potentially a big matrix. can we delete it after all the fits?

**Review by adamgayoso — CHANGES_REQUESTED** [review]

File: src/scib_metrics/utils/_kmeans.py (hunk 2 above)
Comment (martinkim0) [reply src/scib_metrics/utils/_kmeans.py:125]: Done

**Review by adamgayoso — APPROVED** [review]
